backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -202,6 +213,59 @@ public int cancelAll() {
 	  return _repo.cancelAllPendingOrders(_os.getCurrentOrganization());
   }
 
+    @Transactional
+    @AuthorizationConfiguration.RequirePermissionUpdateTest
+    public TestEvent correctTestMarkAsError(String testEventIdStr, String reasonForCorrection) {
+        // The client sends us a TestEvent, we need to map back to the Order.
+        UUID testEventId = UUID.fromString(testEventIdStr);
+        Optional<TestEvent> loadExistingEvent = _terepo.findById(testEventId);
+        if (loadExistingEvent.isEmpty()) {
+            // should this throw?
+            LOG.error("Failed to load TestEvent by id {}", testEventId);
+            return null;
+        }
+        TestEvent event = loadExistingEvent.get();
+        if (event.getCorrectionStatus() != TestCorrectionStatus.ORIGINAL) {
+            LOG.error("TestEvent to be corrected must be in the 'TestCorrectionStatus.ORIGINAL' testEventId {}", testEventId);
+            return null;
+        }
+
+        // todo: should we verify reasonForCorrection is NOT empty? Do we trim()?
+
+        TestOrder order = _repo.findByTestEventId(event.getOrganization(), testEventId);
+        if (order == null) {
+            LOG.error("TestEvent: {} could not load the parent order", testEventId);
+            return null;
+        }
+
+        if (!testEventId.equals(order.getTestEventId())) {
+            LOG.error("TestEvent: parent order {} points to a different TestEvent. Order's TestEventId {} expected {}",
+                    order.getInternalId(), order.getTestEventId(), testEventId);
+            return null;
+        }
```

Comment:
```suggestion
    TestOrder order = event.getTestOrder();
```

Reply:
@TimBest Clarification: So you think these sanity checks are overkill and should be removed?